packages/perseus-core/src/widgets/matcher/matcher-util.ts

```diff
@@ -0,0 +1,138 @@
+import {CoreUtil} from "@khanacademy/perseus-core";
+
+import type {PerseusMatcherWidgetOptions} from "@khanacademy/perseus-core";
+
+// TODO(LEMS-2841): Should be able to remove once getPublicWidgetOptions is hooked up
+type MatcherInfo = {
+    left: ReadonlyArray<string>;
+    right: ReadonlyArray<string>;
+    orderMatters: boolean;
+    problemNum: number | null | undefined;
+};
+
+type MatcherShuffleInfo = {
+    left: ReadonlyArray<string>;
+    right: ReadonlyArray<string>;
+    orderMatters: boolean;
+};
+
+// Note: Previously called cyrb128
```

Comment:
@benchristel Should I link to that stack overflow page in the code? Not sure about attribution within code 🤔

Reply:
Linking to this resource in another PR - https://github.com/Khan/perseus/pull/2215